lib/modules.nix

```diff
@@ -592,11 +592,15 @@ rec {
     { options = setAttrByPath optionName (mkOption {
         visible = false;
       });
-      config.warnings =
+      config.assertions =
         let opt = getAttrFromPath optionName options; in
-        optional opt.isDefined ''
+        optional opt.isDefined {
+          assertion = false;
```

Comment:
You can do `assertion = !opt.isDefined;`

Reply:
Done